src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs

```diff
@@ -118,7 +117,68 @@ private static void OnAutoPlayChanged(DependencyObject sender, DependencyPropert
 
 		#endregion
 
-		#region MediaPlayer Property
+		#region IsFullWindow Property
+
+		public bool IsFullWindow
+		{
+			get { return (bool)GetValue(IsFullWindowProperty); }
+			set { SetValue(IsFullWindowProperty, value); }
+		}
+
+		public static DependencyProperty IsFullWindowProperty { get; } =
+			DependencyProperty.Register(
+				nameof(IsFullWindow),
+				typeof(bool),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(false, OnIsFullWindowChanged));
+
+
+		private static void OnIsFullWindowChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			sender.Maybe<MediaPlayerElement>(mpe =>
+			{
+				mpe.ToogleFullScreen((bool)args.NewValue);
+			});
+		}
+
+		private object _mainContent;
+
+		private void ToogleFullScreen(bool showFullScreen)
+		{
+			if (showFullScreen)
+			{
+				Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
```

Comment:
Why are you dispatching this?

Reply:
Underlying native methods need to run on UI thread

Reply:
Right! fixed